script/upload-to-github.js

```diff
@@ -1,48 +1,68 @@
 if (!process.env.CI) require('dotenv-safe').load()
 
-const GitHub = require('github')
-const github = new GitHub()
-github.authenticate({ type: 'token', token: process.env.ELECTRON_GITHUB_TOKEN })
+const fs = require('fs')
+
+const octokit = require('@octokit/rest')()
+octokit.authenticate({ type: 'token', token: process.env.ELECTRON_GITHUB_TOKEN })
 
 if (process.argv.length < 6) {
   console.log('Usage: upload-to-github filePath fileName releaseId')
   process.exit(1)
 }
+
 const filePath = process.argv[2]
 const fileName = process.argv[3]
 const releaseId = process.argv[4]
 const releaseVersion = process.argv[5]
 
-const targetRepo = releaseVersion.indexOf('nightly') > 0 ? 'nightlies' : 'electron'
+const getHeaders = (filePath, fileName) => {
+  const extension = fileName.split('.').pop()
+  const size = fs.statSync(filePath).size
+  const options = {
+    'json': 'text/json',
+    'zip': 'application/zip',
+    'txt': 'text/plain'
+  }
 
-const githubOpts = {
-  owner: 'electron',
-  repo: targetRepo,
-  id: releaseId,
-  filePath: filePath,
-  name: fileName
+  return {
+    'content-type': options[extension],
```

Comment:
How do you feel about a safe fallback here?

```suggestion
    'content-type': options[extension] || 'application/octet-stream',
```